app/assets/javascripts/places.js

```diff
@@ -2,29 +2,157 @@
 // All this logic will automatically be available in application.js.
 $( document ).ready(function() {
   $(function(){
-    $('[data-toggle="tooltip"]').tooltip();
-  });
+      $('[data-toggle="tooltip"]').tooltip();
+    });    
+
+    $('#modal-add').on('change', '#pictureInput', function(event){
+      console.log('entra');
+      var files = event.target.files;
+      var image = files[0]
+      var reader = new FileReader();
+      reader.onload = function(file) {
+        var img = new Image();
+        console.log(file);
+        img.src = file.target.result;
+        $target = $('#target')
+        $target.html(img);
+        $target.removeClass('photo-frame');
+        $('#modal-add #target img').load(function () {  
+          $(this).addClass('places-images-modal');
+
+        });
+      }
+      reader.readAsDataURL(image);
+      console.log(files);
+    });
+
 
+
+  var lat;
+  var lon;
+  var map;
+  var marker;
+  var geocoder;
+  var mapOptions = {
+                      center: new google.maps.LatLng(0.0, 0.0),
+                      zoom: 2,
+                      mapTypeId: google.maps.MapTypeId.ROADMAP 
+                    };
   
+  if ("geolocation" in navigator) {
+      getLocation();
+    } else {
+      alert("Geolocation is not available")
+    }
+
+  var options = {
+    enableHighAccuracy: true,
+    timeout: 5000,
+    maximumAge: 0
+  };
+
+  function getLocation() {
+    console.log('Getting location...'); 
+    navigator.geolocation.getCurrentPosition(setLocation, onError, options);
+  }
+
+
+  function setLocation (position) {
+    console.log("Got it!");
+    lat = position.coords.latitude;
+    lon = position.coords.longitude;
+    console.log(lat);
+    console.log(lon);
+    $('#latitude-place').val(lat);
+    $('#longitude-place').val(lon);
+    var coord = {
+                  latitude: lat,
+                  longitude: lon
+                };
+
+    $.get('/users/set_coordinates', coord).done(
```

Comment:
I used them later in order to create the navigation link to get directions from googl maps. this way every time the user enter the account I have the value to also do things like painting the map centered in the user location. In any case it is not done well and retrieve some error even though it save the coordinates in the current_user object.